### PR TITLE
feat: have Autocomplete props support composite keys

### PR DESCRIPTION
**Pull request by hein-j**
*Description of changes:*
- add snapshot test for composite keys
- support composite keys in default Autocomplete option `label` values
- support composite keys in Autucomplete option `id` values

* Additional changes:*
- move `buildSelectedRecordsIdSet` for circular dependency
- rename `display-values` to `model-values` to expand scope

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: packages/codegen-ui-react/lib/forms/form-renderer-helper/relationship.ts [hunk 1]
```diff
@@ -66,11 +66,12 @@ export const buildManyToManyRelationshipDataStoreStatements = (
...
-    const { key } = extractModelAndKey(fieldConfigMetaData.valueMappings);
-    if (!key) {
-      throw new InternalError(`Could not identify primary key for ${relatedModelName}`);
+    const { keys } = extractModelAndKeys(fieldConfigMetaData.valueMappings);
+    if (!keys) {
+      throw new InternalError(`Could not identify primary key(s) for ${relatedModelName}`);
     }
-    const relatedModelPrimaryKey = key;
+    // TODO: update for composite
+    const relatedModelPrimaryKey = keys[0];
```
Comment (cwoolum) [inline packages/codegen-ui-react/lib/forms/form-renderer-helper/relationship.ts:74]: Is this supposed to be implemented yet?

File: packages/codegen-ui-react/lib/forms/form-renderer-helper/relationship.ts [hunk 2]
```diff
@@ -1164,11 +1165,12 @@ export const buildHasManyRelationshipDataStoreStatements = (
...
   const linkedDataSet = `${linkedDataName}Set`;
-  const { key } = extractModelAndKey(fieldConfigMetaData.valueMappings);
-  if (!key) {
-    throw new InternalError(`Could not identify primary key for ${relatedModelName}`);
+  const { keys } = extractModelAndKeys(fieldConfigMetaData.valueMappings);
+  if (!keys) {
+    throw new InternalError(`Could not identify primary key(s) for ${relatedModelName}`);
   }
-  const relatedModelPrimaryKey = key;
+  // TODO: update for composite keys
```
Comment (cwoolum) [inline packages/codegen-ui-react/lib/forms/form-renderer-helper/relationship.ts:1172]: Is this supposed to be implemented yet?

File: packages/codegen-ui/lib/generate-form-definition/helpers/model-fields-configs.ts [hunk 3]
```diff
@@ -72,8 +74,15 @@ function getValueMappings({
   // if relationship
   if (field.relationship) {
     const modelName = field.relationship.relatedModelName;
+    // if model, store all keys; else, store field as key
+    const keys =
+      typeof field.dataType === 'object' && 'model' in field.dataType ? allModels[modelName].primaryKeys : [fieldName];
+    const values: StudioFormValueMappings['values'] = [];
```
Comment (cwoolum) [inline packages/codegen-ui/lib/generate-form-definition/helpers/model-fields-configs.ts:80]: We can just use a map here

File: packages/codegen-ui-react/lib/forms/form-renderer-helper/relationship.ts (hunk 1 above)
Comment (hein-j) [reply packages/codegen-ui-react/lib/forms/form-renderer-helper/relationship.ts:74]: Not yet. In the works.

File: packages/codegen-ui-react/lib/forms/form-renderer-helper/relationship.ts (hunk 2 above)
Comment (hein-j) [reply packages/codegen-ui-react/lib/forms/form-renderer-helper/relationship.ts:1172]: Not yet. In the works.

File: packages/codegen-ui/lib/generate-form-definition/helpers/model-fields-configs.ts (hunk 3 above)
Comment (hein-j) [reply packages/codegen-ui/lib/generate-form-definition/helpers/model-fields-configs.ts]: Done

**Review by cwoolum — APPROVED** [review]

**Review by Milan-Shah — APPROVED** [review]

**Pull request merged**